app/assets/javascripts/soundcloud_stream.js.erb

```diff
@@ -1,50 +1,42 @@
-$(document).ready(function(){
-  SC.initialize({
-    client_id: "<%= ENV['SOUNDCLOUD_APP_ID'] %>",
-    redirect_uri: 'http://cryptic-dusk-7305.herokuapp.com/callback'
-  });
-
-  $('#connect').click(function(){
-    var currentUser = new SoundcloudUser
-    currentUser.searchSongs();
-  })
+$('#connect').click(function(){
+  new SoundcloudUser
+})
 
-  $('.track').on('click', function(event){
-    track = new Track({id: event.currentTarget.dataset.trackId, source_id: event.currentTarget.dataset.sourceId})
-    track.audio.play()
-  })
+$('.user-tracks').on('click', '.track', function(event){
+  track = new Track({id: event.currentTarget.dataset.trackId, source_id: event.currentTarget.dataset.sourceId})
+  track.audio.play()
+})
 
-  $('.snippet').on('click', function(event){
-    var snipData = event.currentTarget.dataset;
-    var snippetTrack = new Track({source_id: snipData.sourceId});
-    snippet = new Snippet({track: snippetTrack, startTime: snipData.startTime, endTime: snipData.endTime
-    });
-    snippet.playSnip();
-  })
+$('.user-snippets').on('click', '.snippet', function(event){
+  var snipData = event.currentTarget.dataset;
+  var snippetTrack = new Track({source_id: snipData.sourceId});
```

Comment:
Again, don't take my previous comment as an excuse to reformat everything (like here) .... just don't go crazy with it.